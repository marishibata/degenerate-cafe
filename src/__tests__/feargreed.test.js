import React from 'react';
import { render } from '@testing-library/react';
import FearGreed from '../Components/feargreed';

jest.mock('axios', () => {
  return {
    default: () => {
      return new Promise((resolve, reject) => {
        resolve({
          data: {data: [
            {
              value: 0,
              value_classification: "Extreme Fear"
            }
          ]}
        })
      })
    }
  }
})

describe('fear and greed section', () => {
  it('Renders consistently', async () => {
    const { asFragment } = render(<FearGreed/>);
    expect(asFragment()).toMatchSnapshot();
  })

  it('Successfully renders the component to the screen', async () => {
    const { findByText } = render(<FearGreed/>);
    const result = await findByText("Today's Cryptocurrency Fear And Greed Index");
    expect(result).toBeTruthy();
  })

  it('Successfully fetches F&G index', async () => {
    const { findByText } = render(<FearGreed/>);
    const result = await findByText('Extreme Fear');
    expect(result).toBeTruthy();
  })
})