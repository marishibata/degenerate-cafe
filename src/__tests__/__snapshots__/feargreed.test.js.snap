// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fear and greed section Renders consistently 1`] = `
<DocumentFragment>
  <div
    class="fear-greed"
  >
    <div
      class="title"
    >
      <span>
        Today's Cryptocurrency Fear And Greed Index
      </span>
    </div>
    <div
      class="fear-greed-title"
    >
      How high are we...
    </div>
    <div
      class="fear-greed-number"
    >
      0
    </div>
    <div
      class="fear-greed-title"
    >
      We're feeling...
    </div>
    <div
      class="fear-greed-number"
    >
      Loading
    </div>
  </div>
</DocumentFragment>
`;
